src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -275,6 +278,21 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
                         solution.GetDocumentId(syntax.SyntaxTree),
                         cancellationToken).ConfigureAwait(false);
 
+                    // add import symbol annotations for all the code we may be moving
+                    newDestination = newDestination.WithAdditionalAnnotations(
+                        semanticModel.GetAllDeclaredSymbols(syntax, cancellationToken)
+                        .Select((symbol) =>
+                        {
+                            // prefer the convert because it can handle param and return types
+                            // but use memberType in case of fields/variable declaration.
+                            var type = symbol.ConvertToType(compilation);
+                            if (type.Equals(compilation.ObjectType))
+                            {
+                                type = symbol.GetMemberType();
+                            }
+                            return SymbolAnnotation.Create(type ?? compilation.ObjectType);
```

Comment:
Emm, in which case could type be null here?

Reply:
i also don't really understand what is going on here.  you're finding all the declared symbols... and then doing either COnvertToType or GetMemberType... because?    I don't really get the logic of what's going on here.